vpr/src/pack/cluster.cpp

```diff
@@ -300,6 +300,11 @@ void add_cluster_molecule_candidates_by_highfanout_connectivity(t_pb* cur_pb,
                                                                 const std::multimap<AtomBlockId, t_pack_molecule*>& atom_molecules,
                                                                 const int feasible_block_array_size);
 
+void add_cluster_molecule_candidates_by_attraction_group(t_pb* cur_pb,
+                                                         t_cluster_placement_stats* cluster_placement_stats_ptr,
+                                                         const std::multimap<AtomBlockId, t_pack_molecule*>& atom_molecules,
```

Comment:
could put atom_molecules, cluster_placement_stats_ptr, and the attraction groups data structure in a struct together and pass that around